hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/ProxyWorldUpdater.java

```diff
@@ -282,10 +283,10 @@ public void setupInternalAliasedCreate(@NonNull final Address origin, @NonNull f
      * {@inheritDoc}
      */
     @Override
-    public void finalizeHollowAccount(@NonNull final Address alias) {
-        evmFrameState.finalizeHollowAccount(alias);
+    public void finalizeHollowAccount(@NonNull final Address alias, final long contractNonce) {
+        AccountID hollowAccountId = evmFrameState.finalizeHollowAccount(alias, contractNonce);
         // add child record on merge
-        var contractId = getHederaContractId(alias);
+        var contractId = ConversionUtils.asNumericContractId(hollowAccountId);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** Should we reset the `pendingCreation` to null for this case in which we merge the CREATE2 contract creation into the existing hollow account i.e. if we leave set as it is now could that lead to some unexpected side effect?

Reply:
I think @MiroslavGatsanoga has a point here...In fact, if we set `pendingCreation = null` **before** calling `getHederaContractId(alias)`, everything will work as expected without needing to return the `hollowAccountId` from `finalizeHollowAccount()`.